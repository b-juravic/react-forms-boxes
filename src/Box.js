import React from 'react';

function Box({backgroundColor, width, height, handleClose, id}) {
  let styles = {
    backgroundColor, 
    width,
    height,
    display: 'inline-block'
  }

  // pass id to higher function for reference
  const remove = () => handleClose(id);

  return (
    <div>
      <div style={styles}></div>
      <button onClick={remove}>X</button>
    </div>

  );
}

export default Box;