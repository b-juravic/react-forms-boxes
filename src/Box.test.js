import React from 'react';
import Box from './Box';
import { render } from '@testing-library/react'

// Smoke Test!
it("renders without crashing", function() {
  render(<Box />);
});

//Snapshot Test
it("should match snapshot", function() {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

