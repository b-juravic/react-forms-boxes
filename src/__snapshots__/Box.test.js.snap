// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="display: inline-block;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
